scripts/retrieve_databundle_light.py

```diff
@@ -482,28 +483,33 @@ def get_best_bundles_by_category(
         List of bundles to download
     """
     # dictionary with the number of match by configuration for tutorial/non-tutorial configurations
-    dict_n_matched = {
-        bname: config_bundles[bname]["n_matched"]
-        for bname in config_bundles
-        if config_bundles[bname]["category"] == category
-        and config_bundles[bname].get("tutorial", False) == tutorial
-        and _check_disabled_by_opt(config_bundles[bname], config_enable) != ["all"]
-    }
+    df_matches = pd.DataFrame(columns=["bundle_name", "bundle_size", "n_matched"])
+
+    for bname, bvalue in config_bundles.items():
+        if (
+            bvalue["category"] == category
+            and bvalue.get("tutorial", False) == tutorial
+            and _check_disabled_by_opt(bvalue, config_enable) != ["all"]
+        ):
+            df_matches.loc[bname] = [
+                bname,
+                len(bvalue["countries"]),
+                bvalue["n_matched"],
+            ]
 
-    returned_bundles = []
+    df_matches["neg_bundle_size"] = -df_matches["bundle_size"]
+    df_matches.sort_values(
+        by=["n_matched", "neg_bundle_size"], inplace=True, ascending=False
+    )
 
-    # check if non-empty dictionary
-    if dict_n_matched:
-        # if non-empty, then pick bundles until all countries are selected
-        # or no more bundles are found
-        dict_sort = sorted(dict_n_matched.items(), key=lambda d: d[1])
+    bname_list = list(df_matches["bundle_name"])
+    returned_bundles = []
 
+    if bname_list:
         current_matched_countries = []
         remaining_countries = set(country_list)
 
-        for d_val in dict_sort:
-            bname = d_val[0]
-
+        for bname in bname_list:
```

Comment:
Great job @Emre-Yorat89 :D
The part above this line I believe is basically perfect!

The following may be slightly revised but I think we are close to done.

Proposed changes are:
- here we no more need to define b_name_list, you can delete row 505; accordingly, line 512 may be revised in checking whether the dataframe is not empty: not df_matches.empty
- then the loop over b_name in b_name_list can be revised into a loop over the rows of the dataframe: `for bname, row in df_matches.iterrows():`

With these changes, the approach should work as is.
Could you try if it works?

Reply:
Do you observe unexpected behaviors?
I have the function in a couple of different countries and it seems it works okay. One thing I have noticed when downloading data for european countries

Reply:
Great @emre :D
Tomorrow I'll go through the review but I think it is ready to go!
Agree with your comment: we miss a european cutout and an Oceania cutout. Tommaso was interested in the European one, but not sure on its status and the Oceania is yet missing.
Btw the Pr is ready I think:D tomorrow will be merged most likely :)